pkg/cloud/packet/client.go

```diff
@@ -215,34 +226,39 @@ func (p *Client) GetDeviceByTags(project string, tags []string) (*packngo.Device
 
 // CreateIP reserves an IP via Packet API. The request fails straight if no IP are available for the specified project.
 // This prevent the cluster to become ready.
-func (p *Client) CreateIP(namespace, clusterName, projectID, facility string) (net.IP, error) {
-	req := packngo.IPReservationRequest{
-		Type:                   packngo.PublicIPv4,
+func (p *Client) CreateIP(ctx context.Context, namespace, clusterName, projectID, facility string) (net.IP, error) {
+	failOnApprovalRequired := true
+	req := metal.IPReservationRequestInput{
+		Type:                   "public_ipv4",
```

Comment:
openapitools-generator doesn't make enums?! 😭

Reply:
This is a [bug in the Go generator](https://github.com/OpenAPITools/openapi-generator/issues/2360).  In `metal-java` [the IP type is a proper enum](https://github.com/equinix-labs/metal-java/blob/main/equinix-openapi-metal/src/main/java/com/equinix/openapi/metal/v1/model/IPReservation.java#L177).  We could try patching the spec in `metal-go` to move the enum to a named component and see if that fixes it.

Reply:
I filed a `metal-go` issue to look into `enum` generation: https://github.com/equinix-labs/metal-go/issues/124
There is the potential for unintended consequences with `enum`, so I lean towards keeping this PR as-is, and coming back later to use proper enums.